plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -205,24 +212,83 @@ const addToCartWithOptionsStore = store(
 				// woocommerce store is public.
 				yield import( '@woocommerce/stores/woocommerce/cart' );
 
-				const { actions } = store< WooCommerce >(
-					'woocommerce',
-					{},
-					{ lock: universalLock }
-				);
+				const {
+					productId,
+					quantity,
+					selectedAttributes,
+					productType,
+					groupedProductIds,
+				} = getContext< Context >();
 
-				const { productId, quantity, selectedAttributes } =
-					getContext< Context >();
-				const product = wooState.cart?.items.find(
-					( item ) => item.id === productId
-				);
-				const currentQuantity = product?.quantity || 0;
+				if (
+					productType === 'grouped' &&
+					groupedProductIds.length > 0
+				) {
+					// Handle grouped products
+					const form = event.target as HTMLFormElement;
+					const quantityInputs = form.querySelectorAll(
+						'.input-text.qty.text'
+					);
```

Comment:
I wonder if instead of querying the input HTML elements here, it might be better to have each product's quantity saved in the store. So `quantity`, instead of being an int, it could be an object matching `productId => quantity`.

I'm fine leaving this for a follow-up PR if you prefer, as the current approach works, but I feel that saving the quantities in the store is a more iAPI-_ish_ approach. What do you think?

Reply:
Great idea, thank you. I've tried this in https://github.com/woocommerce/woocommerce/pull/58002/commits/9a60a1eba5a522f7e4429633339f10845e468350. I'm hoping I've understood correctly - let me know otherwise!